### PR TITLE
ci/cleanup: clean up leaked S3 buckets and SQS queues

### PR DESCRIPTION
**Pull request by benesch**
The new S3 test can leak buckets and SQS queues; when enough of these
pile up, we hit AWS resource limits and all CI jobs start failing.
Extend the Kinesis cleanup job to clean up leaked S3 buckets and SQS
queues too. (We can't rely on testdrive to clean up after itself in all
cases if e.g. a build agent dies while running a job.)

This commit tweaks testdrive to force usage of a "testdrive-" prefix in
bucket/queue names, which brings those resources in line with Kafka
topics and Kinesis streams created by testdrive, and also makes it easy
to identify them in the cleanup job.

### CODE REVIEW

File: test/testdrive/esoteric/s3-gzip.td [hunk 1]
```diff
@@ -23,7 +23,7 @@ b2
 b3
 
 > CREATE MATERIALIZED SOURCE s3_all_none
-  FROM S3 OBJECTS FROM SCAN BUCKET '${bucket}'
+  FROM S3 OBJECTS FROM SCAN BUCKET 'testdrive-${bucket}'
```
Comment (cirego) [inline test/testdrive/esoteric/s3-gzip.td:26]: Minor nit: this bucket name will now be "testdrive-materialize-ci-testdrive-none-${testdrive.seed}". Perhaps we should remove the long prefixes from the testdrive bucket names now that testdrive formats the bucket name with the "testdrive" prefix?

**Review by cirego — APPROVED** [review]
Looks great to me -- thanks for picking this up!

File: test/testdrive/esoteric/s3-gzip.td (hunk 1 above)
Comment (benesch) [reply test/testdrive/esoteric/s3-gzip.td:26]: Oops, yeah, meant to do that! Thanks for the reminder.

**Comment by benesch** [comment]
Ugh, IAM permissions. https://github.com/MaterializeInc/infra/pull/895

**Review by cirego — APPROVED** [review]

**Review by quodlibetor — APPROVED** [review]

**Pull request merged**